### PR TITLE
Add an example for the usage of different semirings

### PR DESCRIPTION
**Pull request by FlorentinD**
.. as requested in #82

this is just a starting point.
I tried to create a very minimalistic example. 
Maybe someone has some suggestions for better/more explanations in the example.

### CODE REVIEW

File: examples/src/org/ejml/example/ExampleSemiRings.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+package org.ejml.example;
+
+import org.ejml.data.DMatrixSparseCSC;
+import org.ejml.ops.DMonoids;
+import org.ejml.ops.DSemiRing;
+import org.ejml.ops.DSemiRings;
+import org.ejml.sparse.csc.CommonOpsWithSemiRing_DSCC;
+
+/**
+ * Example showing the usage of semirings in the context of graphs as for instance used in GraphBLAS
```
Comment (lessthanoptimal) [inline examples/src/org/ejml/example/ExampleSemiRings.java]: Could you explain very briefly which problem you are solving in the description. Also might want to change the class name if possible. E.g.
`Example showing how to find the lowest cost path in a graph using semirings`

Then `ExampleGraphBestPath`

More people will understand that language and might realize this is useful for them.

File: examples/src/org/ejml/example/ExampleSemiRings.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        System.out.println("Node 3 can be reached from node 0: " + (outputVector.get(0, 3) == 1));
+        System.out.println("Node 1 can be reached from node 0: " + (outputVector.get(0, 1) == 1));
+
+
+        // Add node 3 to the start nodes
+        startNodes.set(0, 3, 1);
+
+        // Find the path with the minimal cost
+        CommonOpsWithSemiRing_DSCC.mult(startNodes, adjacencyMatrix, outputVector, min_times, null, null);
```
Comment (lessthanoptimal) [inline examples/src/org/ejml/example/ExampleSemiRings.java]: How is the minimum cost path defined in this situation. Sum of cost across the entire path?

**Review by lessthanoptimal — COMMENTED** [review]
A couple of comments about documentation.

File: examples/src/org/ejml/example/ExampleSemiRings.java (hunk 2 above)
Comment (FlorentinD) [reply examples/src/org/ejml/example/ExampleSemiRings.java]: In this case its just the cost of the relationship between the nodes, as we are just doing one iteration of bfs.
As we are using the `MIN_TIMES` semi-ring its weight in startNodes vector * relationship-cost (but the startNodes entry are both `1` for simplicity).

I hope my added comments are understandable :)

**Comment by FlorentinD** [comment]
@lessthanoptimal I addressed your comments and hope its now more understand-able

**Comment by lessthanoptimal** [comment]
thanks! Approving. Hopefully you can pick this up again in the future and add more functionality!

**Review by lessthanoptimal — APPROVED** [review]
Approved but could you squash first before we merge?

**Comment by FlorentinD** [comment]
Squashed.
I plan to add more functionality in general like masks for sparse matrices (which is on my fork for now). However I want to focus on my thesis for now and create PRs in January.

**Pull request merged**